Updates to presets / custom toggle and segmented selection

LevinMedia opened the pull request:
**presets / custom toggle:**

- changed the font color to `$woocommerce` for the inactive half of the button
- Added a subtle hover effect to the inactive half of the button
- Decreased the height by 2px

**Segmented Selection**

- Updated the padding around the purple square that indicates a preset date range, or comparison period is selected
- Increased the font weight on the selected label

**Update & Reset buttons**

- Changed the width to 40% 
- Changed the height to 34px 
-Adjusted margin between buttons to .75em

Had to force the last two with !important - if anyone has a better way of doing that 🙌 

**Before**

<img width="473" alt="screen shot 2018-08-22 at 7 08 12 pm" src="https://user-images.githubusercontent.com/4500952/44501675-1b8c4f80-a643-11e8-9a10-741a048b941d.png">

<img width="465" alt="screen shot 2018-08-22 at 7 08 29 pm" src="https://user-images.githubusercontent.com/4500952/44501674-1af3b900-a643-11e8-8b55-220e9c7d8430.png">

**After**

<img width="464" alt="screen shot 2018-08-22 at 7 11 39 pm" src="https://user-images.githubusercontent.com/4500952/44501723-44ace000-a643-11e8-9334-4da1bf58ec55.png">

<img width="461" alt="screen shot 2018-08-22 at 7 12 10 pm" src="https://user-images.githubusercontent.com/4500952/44501722-44ace000-a643-11e8-87db-925abf719963.png">

**Not included in this PR**

Will open up a new issue with the following tasks for someone who's got moar skillz than me:

- The "reset" button on the custom tab should be using the secondary button style outlined in p6riRB-3hq-p2
- Need to fix a width somewhere that's rendering a horizontal scroll bar at the bottom of the custom tab (visible in the screen shots above)
- Update focus style on custom tab month switcher buttons
- Update error state and include tool tip per p6riRB-3hq-p2 for manual date entry on the custom tab
- Segmented selections in the left hand column of the presets tab jump 1px to the right when selected (something to do with the focus outline)

LevinMedia committed:
Updates to segmented selection and presets / custom toggle

jameskoster commented:
What's that 3/4px horizontal line beneath the update/reset buttons? 🤔

justinshreve commented on client/components/filters/date/style.scss:
```diff
@@ -37,11 +37,17 @@
 	outline: none;
 	border: 1px solid $woocommerce;
-	padding: 10px;
+	padding: 9px;
```
We have some variables for padding/spacing: 

https://github.com/woocommerce/wc-admin/blob/master/client/stylesheets/abstracts/_variables.scss

Should this be `$gap-smaller` (8px) instead of 9px?

LevinMedia replied:
Sounds good - I'm also going to add a `$gap-larger` at 36px to address the comment from @ryelle below

justinshreve commented on client/components/filters/date/style.scss:
```diff
@@ -95,6 +101,7 @@
-	margin: 0 1em;
+	width: 40%;
+	height: 34px !important;
+	margin: 0 0.75em !important;
```
Looks like 0.75em is 12px, so this could use `$gap-small`

justinshreve commented:
> What's that 3/4px horizontal line beneath the update/reset buttons?

It's a horizontal scroll bar that is showing up for some reason (flagged as an issue above).

justinshreve commented on client/components/filters/date/style.scss:
```diff
@@ -95,6 +101,7 @@
-	width: 50%;
-	margin: 0 1em;
+	width: 40%;
+	height: 34px !important;
```
These rules work without needing to use the !important.

```
.woocommerce-filters-date__button {
	justify-content: center;
	width: 40%;
	&.components-button {
		margin: 0 $gap-small;
		&.is-button {
			height: 34px;
			line-height: 32px;
		}
	}
}
```

ryelle replied:
It is a matter of specificity, so that would work, but it's a little unclear that those styles always apply - it would be better to keep everything in one rule-block. We could nest it in `.woocommerce-filters-date__button-group`, and add the extra specificity of `is-button`:

```
.woocommerce-filters-date__button-group {
	…
	.woocommerce-filters-date__button.is-button {
		justify-content: center;
		width: 40%;
		height: 34px !important;
		margin: 0 0.75em !important;
	}
}
```

justinshreve commented on client/components/segmented-selection/style.scss:
```diff
@@ -17,7 +17,7 @@
 .woocommerce-segmented-selection__label {
 	background-color: $core-grey-light-100;
-	padding: 12px 12px 12px 24px;
+	padding: 12px 12px 12px 36px;
```
Maybe `padding: $gap-small $gap-small $gap-small $gap-largest;` instead?

ryelle replied:
`$gap-largest` is 40px, so it would be a slightly different style if that was used.

emdashcodes reviewed (commented):
A few comments about using our spacing variables, but the improvements are looking good in testing 👍.

LevinMedia committed:
style updates per review

LevinMedia commented:
Made updates per the feedback from the last commit. I added a `$gap-larger` at 36px to the variables. Also changed the radius on the presets / custom toggle from 5px to 4px.

emdashcodes reviewed (approved):
LGTM 👍
